src/resolve/resolved-dns-transaction.c

```diff
@@ -3333,10 +3333,18 @@ static int dnssec_validate_records(
                         }
                 }
 
+                if (result == DNSSEC_UNSUPPORTED_ALGORITHM) {
+                        r = dns_answer_move_by_key(validated, &t->answer, rr->key, 0, NULL);
```

Comment:
can you add a comment refefencing https://datatracker.ietf.org/doc/html/rfc6840#section-5.2 please?

Reply:
trivial change to make, so just pushed so we can get the fix merged for the next release